packages/thirdweb/src/transaction/prepare-contract-call.ts

```diff
@@ -193,11 +213,27 @@ export function prepareContractCall<
           preparedM = await preparedMethodPromise();
         }
 
+        // just return the fn sig directly -> no params
         if (preparedM[1].length === 0) {
```

Comment:
I wouldn't append the extra data here. Instead i would append it inside `encode(transaction)`

Reply:
<img width="600" alt="image" src="https://github.com/user-attachments/assets/0f11fec5-afd3-4f96-9273-b4a912ddf678">

@joaquim-verges joa
do you mean we should do the concatHex for the extra calldata in here?